Fix #3557 Merge profile chooser fragment into 2 fargments

priyatanu opened the pull request:
<!-- READ ME FIRST: Please fill in the explanation section below and check off every point from the Essential Checklist! -->
## Explanation

Merged 2 files of profile_chooser_fragment.xml into one profile_chooser_fragment.xml in layout/profile_chooser_fragment.xml
But for the larger screen size I found that sw600dp profile_chooser_fragment is different to mobile screen layout. So I have merged 2 sw600dp modes profile_chooser_fragment into 1 profile_chooser_fragment.xml inside sw600dp folder
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->

## Essential Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [ ] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [ ] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [ ] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [ ] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [ ] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only
<!-- Delete these section if this PR does not include UI-related changes. -->
If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

priyatanu committed:
Merge profile_chooser_fragment into one xml across differnt screen sizes

priyatanu commented:
![Profile_Chooser1](https://user-images.githubusercontent.com/18245363/152678331-d7e9b0d9-81c1-49a8-91b1-ee0b0145134e.PNG)

priyatanu commented:
![ProfileChooser2](https://user-images.githubusercontent.com/18245363/152678337-1ef777dc-bc77-4e24-8f37-a70c172dfd13.PNG)

priyatanu commented:
![ProfileChooser3](https://user-images.githubusercontent.com/18245363/152678461-2e8a213e-9bbf-4f1c-b44b-438543d9950a.PNG)

priyatanu commented:
![ProfileChooser4](https://user-images.githubusercontent.com/18245363/152678466-bdb6b669-82ee-497b-80f3-10ef414de3e0.PNG)

priyatanu committed:
Rename layout attributes

priyatanu commented:
PTAL @rishidyno , @ARJUPTA ,

Kindly review .

I made mistakes while rebasing the old PR while trying to open it https://github.com/oppia/oppia-android/pull/4143

So I created a new PR. sorry for the confusion

rishidyno commented on app/src/main/res/values/dimens.xml:
```diff
@@ -457,6 +457,12 @@
   <dimen name="profile_chooser_fragment_profile_recycler_view_margin_top">36dp</dimen>
   <dimen name="profile_chooser_profile_view_view_margin_top">60dp</dimen>
 
```
Please remove extra line.

rishidyno commented on app/src/main/res/values/dimens.xml:
```diff
@@ -457,6 +457,12 @@
+  <dimen name="profile_chooser_fragment_layout_margin_end">32dp</dimen>
+  <dimen name="profile_chooser_fragment_layout_height">128dp</dimen>
+  <dimen name="profile_chooser_fragment_layout_margin_bottom">32dp</dimen>
+
```
Ditto. Remove this extra line.

rishidyno reviewed (changes requested):
@priyatanu Please add SS in a table take reference [here](https://github.com/oppia/oppia-android/pull/3644)

rishidyno commented:
Also, I can see that you are taking SS separately, use emulator SS feature to Take SS. 
![image](https://user-images.githubusercontent.com/78549195/153178957-3e4d7a5b-68fc-4681-b7ca-370c5266c98d.png)

Arjupta commented on app/src/main/res/layout-land/profile_chooser_fragment.xml:
```diff
@@ -1,121 +0,0 @@
-          app:layout_constraintEnd_toStartOf="@+id/profile_chooser_language_icon"
-          app:layout_constraintTop_toTopOf="parent" />
-
-        <TextView
-          android:id="@+id/profile_select_text"
-          style="@style/Heading1ViewStart"
-          android:layout_marginStart="36dp"
```
The style attribute here differs from the one in portrait xml (mobile). you forgot to resolve this.

Arjupta commented on app/src/main/res/layout-sw600dp/profile_chooser_fragment.xml:
```diff
@@ -108,7 +108,8 @@
       <TextView
         style="@style/Heading1ViewCenter"
-        android:layout_marginBottom="32dp"
+        android:layout_marginBottom="@dimen/profile_chooser_fragment_layout_margin_bottom"
+        android:gravity="center"
         android:minHeight="48dp"
```
can you also explain why a new **gravity** attribute has been introduced?

Arjupta commented on app/src/main/res/values-sw600dp-land/dimens.xml:
```diff
@@ -25,6 +25,14 @@
+  <!-- Profile Chooser Fragment -->
+  <dimen name="profile_chooser_fragment_layout_margin_start">96dp</dimen>
+  <dimen name="profile_chooser_fragment_layout_margin_bottom">28dp</dimen>
+  <dimen name="profile_chooser_fragment_layout_margin_end">96dp</dimen>
+  <dimen name="profile_chooser_fragment_margin_top_profile_already_added">0dp</dimen>
+  <dimen name="profile_chooser_fragment_margin_top_profile_not_added">84dp</dimen>
+
```
If the `profile_chooser_fragment_margin_top_profile_already_added` and `profile_chooser_fragment_margin_top_profile_not_added` are already defined in the portrait xml (mobile). Then we do not need to denfine it here again.

Arjupta reviewed (changes requested):
@priyatanu PTAL at the requested changes.

priyatanu commented:
open

priyatanu commented:
Hi, 
@rishidyno @ARJUPTA ,

I got extremely bsy with work. So the pull request is stale. I have addressed the comments in my new Pull request
- https://github.com/oppia/oppia-android/pull/4282,

Extremely sorry for the inconvenice
